.scripts/dip721/deploy-nft.sh

```diff
@@ -17,6 +17,23 @@ if [[ "$MODE" == "reinstall" ]]; then
   OPTIONAL+=(--mode reinstall)
 fi
 
+dfx canister --no-wallet \
+  create nft --controller "$OWNER_PRINCIPAL_ID"
+
+CREATED_NFT_CANISTER_ID=$(dfx canister id nft)
+
+printf "🤖 The created NFT Canster id is %s" "$CREATED_NFT_CANISTER_ID"
+
+# Note that the controller settings is important
+# you should have at least an identity you control
+# and the canister itself as controllers
+dfx canister --no-wallet \
+  --network "$NETWORK" \
+  update-settings \
+    --controller "$OWNER_PRINCIPAL_ID" \
+    --controller "$CREATED_NFT_CANISTER_ID" \
+  "$CREATED_NFT_CANISTER_ID"
```

Comment:
- We should wrap this with a check to see if the nft canister is already created
-  is the --controller necessaryin the canister create? 
- read the canister id with --network flag to support local and ic deploys
```sh
CREATED_CANISTER_NFT_ID=$(dfx canister --network $NETWORK id $NFT)
if [[ ! $? -eq 0 ]]; then
    dfx canister --no-wallet create nft

    CREATED_NFT_CANISTER_ID=$(dfx canister --network $NETWORK id $NFT)

    printf "🤖 The created NFT Canster id is %s" "$CREATED_NFT_CANISTER_ID"

    # Note that the controller settings is important
    # you should have at least an identity you control
    # and the canister itself as controllers
    dfx canister --no-wallet \
      --network "$NETWORK" \
      update-settings \
      --controller "$OWNER_PRINCIPAL_ID" \
      --controller "$CREATED_NFT_CANISTER_ID" \
      "$CREATED_NFT_CANISTER_ID"
fi

Reply:
Nice one, I'll add the missing network!

So, the controller is really just being too verbose to communicate requirements, apart from that there's no other advantage.

For the wrapping, mind sending a PR on that? I think the use-case here, which is an implementation example, is already a bit too much.

Reply:
@ozwaldorf updated, the remaining requests please address that separately. Confirm if everything's ok asap, thanks ;)

Reply:
@ozwaldorf fair enough, I'll see what I can do

Reply:
@ozwaldorf updated!